### PR TITLE
Remove internal use of gpu_id.

### PR DESCRIPTION
**Pull request by trivialfis**
Device communicator is still using integer ordinal, which will be changed by upcoming PRs.

### CODE REVIEW

**Comment by hcho3** [comment]
Why are we dropping gtest from the CI?

**Comment by trivialfis** [comment]
@hcho3 windows build is failing with cmake find gtest. (same error on master branch). Since the test is building the Python package, I figure we don't need to build c++ tests as we are not running it.

**Comment by hcho3** [comment]
Are we not running gtest on Windows CI?

**Comment by trivialfis** [comment]
only on buildkite.

**Comment by hcho3** [comment]
Ah, got it

File: src/common/host_device_vector.cu [hunk 1]
```diff
@@ -138,7 +138,7 @@ class HostDeviceVectorImpl {
     } else {
       auto ptr = other->ConstDevicePointer();
       SetDevice();
-      CHECK_EQ(this->DeviceIdx(), other->DeviceIdx());
+      CHECK_EQ(this->Device().ordinal, other->Device().ordinal);
```
Comment (hcho3) [inline src/common/host_device_vector.cu:141]:
```suggestion
      CHECK(this->Device() == other->Device());
```

File: src/common/host_device_vector.cu [hunk 2]
```diff
@@ -156,24 +156,25 @@ class HostDeviceVectorImpl {
...
     if (device_ == device) { return; }
-    if (device_ >= 0) {
+    if (device_.IsCUDA()) {
       LazySyncHost(GPUAccess::kNone);
     }
 
-    if (device_ >= 0 && device >= 0) {
-      CHECK_EQ(device_, device) << "New device ordinal is different from previous one.";
+    if (device_.IsCUDA() && device.IsCUDA()) {
+      CHECK_EQ(device_.ordinal, device.ordinal)
```
Comment (hcho3) [inline src/common/host_device_vector.cu:166]: Is it possible for two `DeviceOrd` objects to be different when they have the same positive value in the `ordinal` field ?

File: src/predictor/gpu_predictor.cu [hunk 3]
```diff
@@ -1048,11 +1044,8 @@ class GPUPredictor : public xgboost::Predictor {
     if (tree_weights != nullptr) {
       LOG(FATAL) << "Dart booster feature " << not_implemented;
     }
-    CHECK(!p_fmat->Info().IsColumnSplit())
-        << "Predict contribution support for column-wise data split is not yet implemented.";
```
Comment (hcho3) [inline src/predictor/gpu_predictor.cu:1052]: Is SHAP support available for column-wise split data now?

**Review by hcho3 — APPROVED** [review]

File: src/common/host_device_vector.cu (hunk 2 above)
Comment (trivialfis) [reply src/common/host_device_vector.cu:166]: Not possible at the moment. Unless there's new device type other than CUDA.

File: src/predictor/gpu_predictor.cu (hunk 3 above)
Comment (trivialfis) [reply src/predictor/gpu_predictor.cu:1052]: The check was removed between PR rebases. I will bring it back.

**Pull request merged**